projects/com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/AsmFactory.java

```diff
@@ -928,6 +1499,24 @@ void addFrameSlot(String reg, Type type) {
         }
     }
 
+    private static PrimitiveKind getPrimitiveKind(Argument arg) {
+        PrimitiveKind primitiveKind;
+        if (arg.getType() instanceof PrimitiveType) {
+            primitiveKind = ((PrimitiveType) arg.getType()).getPrimitiveKind();
+        } else if (arg.getType() instanceof PointerType) {
```

Comment:
I believe that this is necessary ;-) but help me understand:
if an argument type is a pointer, this returns the pointee type. why?
(maybe the method name is just misleading)